src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs

```diff
@@ -8,6 +8,32 @@
 
 namespace Microsoft.ML
 {
+    /// <summary>
+    /// Specifies input and output column names for a transformation.
+    /// </summary>
+    public sealed class InputOutputColumnPair
+    {
+        /// <summary>
+        /// Name of the column to transform. If set to <see langword="null"/>, the value of the <see cref="OutputColumnName"/> will be used as source.
+        /// </summary>
+        public readonly string InputColumnName;
```

Comment:
> public readonly string InputColumnName [](start = 7, length = 39)

I'm slightly confuse.
We get rid of ColumnOptions because they were immutable, and now we add another immutable class...

Reply:
Will invite you to check with Tom about this, I am just executing what he asked

---
In reply to: [268333577](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268333577) [](ancestors = 268333577)